.circleci/config.yml

```diff
@@ -27,7 +27,11 @@ jobs:
           git checkout $CIRCLE_BRANCH || git checkout master
           export WEBAPP_GIT_COMMIT=$(git rev-parse HEAD)
           echo "$WEBAPP_GIT_COMMIT"
-          curl -f -o ./dist.tar.gz https://releases.mattermost.com/mattermost-webapp/commit/${WEBAPP_GIT_COMMIT}/mattermost-webapp.tar.gz && mkdir ./dist && tar -xvf ./dist.tar.gz -C ./dist --strip-components=1 || make node_modules test build
+          curl -f -o ./dist.tar.gz https://releases.mattermost.com/mattermost-webapp/commit/${WEBAPP_GIT_COMMIT}/mattermost-webapp.tar.gz && mkdir ./dist && tar -xvf ./dist.tar.gz -C ./dist --strip-components=1 || echo "curl failed" && export CURL_FAILED=1
```

Comment:
What do you think about using `--retry` instead?

Reply:
@hanzei This checks if the build already exists and if not it builds it in the next step. What to use retry for? :/

Reply:
did not get the retry.

Reply:
I don't think I understand the script correctly. Why should a build already exist? I suppose the pipeline is (mostly) just run once for every commit.

Reply:
the way it works is like this:
1. we checkout the `master` or the `release-branch` (if is in the release branch) or the same branch name as the server (if exist) 
2. we get the commit hash
3. we try to download that commit hash, because the pipeline for webapp build that and push to s3
4. if that hash does not exist is because most probably the webapp is building that one or the webapp pipeline failed to build that hash.
5. if we cannot download the existing webapp package for that hash we build that in this pipeline

most of the cases we can download that and doing that we make a little bit faster this pipeline.

Reply:
this is the way we are doing for quite some time (more then 2 years i guess)

Reply:
Thank you very much for the explanation. Makes sense to me.